### PR TITLE
[HIPTEST]common changes for unittest

### PR DESCRIPTION
**Pull request by SarbojitAMD**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -561,8 +561,12 @@ if(${RUN_HIT} EQUAL 0)
     include_directories(${HIP_SRC_PATH}/tests/src)
     hit_add_directory_recursive(${HIP_SRC_PATH}/tests/src "directed_tests")
 
+    # Add unit tests
+    include_directories(${HIP_SRC_PATH}/tests/unittests)
```
Comment (mangupta) [inline CMakeLists.txt:565]: Naming the folder as tests/unittests sounds redundant. Maybe just go with tests/unit?

File: tests/unittests/test_common.h [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+#define KGRN "\x1B[32m"
+#define KYEL "\x1B[33m"
+#define KBLU "\x1B[34m"
+#define KMAG "\x1B[35m"
+#define KCYN "\x1B[36m"
+#define KWHT "\x1B[37m"
+
+#define passed()                                                                                   \
+    printf("%sPASSED!%s\n", KGRN, KNRM);                                                           \
+    exit(0);
```
Comment (mangupta) [inline tests/unittests/test_common.h:56]: should this be ```return true``` instead?

File: tests/unittests/test_common.h [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+    printf("warn: TEST WARNING\n%s", KNRM);
+
+#define skipped() printf("%sSkipped subtest %s%s\n",KYEL,__FUNCTION__,KNRM);
+
+#define HIPCHECK(error)                                                                            \
+    {                                                                                              \
+        hipError_t localError = error;                                                             \
+        if ((localError != hipSuccess) && (localError != hipErrorPeerAccessAlreadyEnabled)) {      \
+            printf("%serror: '%s'(%d) from %s at %s:%d%s\n", KRED, hipGetErrorString(localError),  \
+                   localError, #error, __FUNCTION__, __LINE__, KNRM);                                  \
```
Comment (mangupta) [inline tests/unittests/test_common.h:77]: Can't this be merged into the ```failed``` line similar to ```HIPASSERT```?

File: CMakeLists.txt (hunk 1 above)
Comment (SarbojitAMD) [reply CMakeLists.txt]: Done

File: tests/unittests/test_common.h (hunk 2 above)
Comment (SarbojitAMD) [reply tests/unittests/test_common.h]: Removed exit(0) as we will have return in main anyway.

File: tests/unittests/test_common.h (hunk 3 above)
Comment (SarbojitAMD) [reply tests/unittests/test_common.h]: done

**Pull request merged**